### ISSUE TITLE
Responsive height

### ISSUE THREAD

**Issue by erperejildo** [serenader2014/flutter_carousel_slider#106]
How can I get a dynamic height for each carousel element?

**Comment by jacksonw765** [comment]
What I did was create a formula that's based on the device height. You can do this using linear regression.

**Comment by erperejildo** [comment]
yeah, I could do that but I was wondering if it was possible using the package itself

**Comment by serenader2014** [comment]
Hi, sorry for the late response.

Currently there is no way to define a dynamic height for each carousel item. But I think it can be done by creating a dynamic height container in the carousel item.

**Comment by ovib98** [comment]
> 
> 
> What I did was create a formula that's based on the device height. You can do this using linear regression.

Hi, do you think there is a similar solution for [my case](https://stackoverflow.com/questions/61207980/create-a-flutter-carousel-slider-with-dynamic-heigth-for-each-page/61211296#61211296)? I would like to adapt the the height according to the content and not necessarily based on the device height.

**Comment by erperejildo** [comment]
> Hi, sorry for the late response.
> 
> Currently there is no way to define a dynamic height for each carousel item. But I think it can be done by creating a dynamic height container in the carousel item.

I tried that but it didn't work. Do you have any specific idea on mind?

**Comment by finxster** [comment]
Do you guys think that this feature will be available here soon? I'm also facing this issue, but instead trying a workaround, I'd like a more definitive solution...

Thank you in advance!

**Comment by serenader2014** [comment]
Unfortunately, I think it is not easy to implement this feature. Because we use `PageView` to display the carousel, but `PageView` doesn't allow us to have a responsive children's height, that being said, you can only create a fixed height carousel item for all the slider :(

**Comment by finxster** [comment]
Thanks @serenader2014. I'll try an alternative workaround then.

**Comment by happyfloat** [comment]
Does anybody has an example how to do it / calculate it?

**Comment by happyfloat** [comment]
I moved to this hacky solution now I found elsewhere instead of carousel:
```dart
                            Column(
                                mainAxisSize: MainAxisSize.min,
                                children: <Widget>[
                                  SingleChildScrollView(
                                    scrollDirection: Axis.horizontal,
                                    child: Row(
                                      mainAxisAlignment: MainAxisAlignment.start,
                                      crossAxisAlignment: CrossAxisAlignment.start,
                                      children: []
                                    ),
                                  ),
                                ],
                              ),
```

But now my main problem is, that it does not snap to my items in the row... maybe I need a custom scroll handler now to fix this.

**Comment by amoscalamida** [comment]
@serenader2014 A recent package has extended the PageView to be expandable in height (https://pub.dev/packages/expandable_page_view/) 
Any chance, that this would be integrated in a future version of carousel_slider?
Adaptive height is the only thing missing on this otherwise amazing package 🥇

**Comment by PawelZaw** [comment]
Maybe try with dynamic changing height in StatefulWidget. The example below is for changing the width. Sorry for formatting:


`import 'package:carousel_slider/carousel_slider.dart';

class HomeBannerDesktop extends StatefulWidget {
  const HomeBannerDesktop({Key key}) : super(key: key);

  @override
  _HomeBannerDesktopState createState() => _HomeBannerDesktopState();
}

class _HomeBannerDesktopState extends State<HomeBannerDesktop> {
  double _width = 370;
  int _current = 0;
  static const double _fontSize = 60;

  List widths = [370, 700, 600];
  List<Widget> textSliders = [
    SliderItem(
      text: "text text.",
      width: 370,
    ),
    SliderItem(
        text: "text text text text.",
      width: 700,
    ),
    SliderItem(
      text: "text text text.",
      width: 600,
    ),
  ];

  @override
  Widget build(BuildContext context) {
    final size = MediaQuery.of(context).size;

    return Container(
      width: size.width,
      color: Colors.amber,
      padding: EdgeInsets.only(
        top: 80,
        bottom: 40,
      ),
      child: Center(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Padding(
              child: Container(
                width: 1050,
                child: Row(
                  mainAxisSize: MainAxisSize.min,
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    Container(
                      padding: EdgeInsets.only(
                        right: 20,
                      ),
                      child: Text(
                        "Text:",
                       
                      ),
                    ),
                     AnimatedContainer(
                      duration: Duration(
                        milliseconds: 1000,
                      ),
                      child: Container(
                        width: widths[_current],
                        child: Column(
                          children: [
                            CarouselSlider(
                              items: textSliders,
                              options: CarouselOptions(
                                  autoPlay: true,
                                  autoPlayInterval: const Duration(seconds: 2),
                                  autoPlayAnimationDuration: Duration(
                                    milliseconds: 800,
                                  ),
                                  viewportFraction: 1,
                                  height: 80,
                                  scrollDirection: Axis.vertical,
                                  scrollPhysics: ScrollPhysics(),
                                  onPageChanged: (index, reason) {
                                    setState(() {
                                      _current = index;
                                    });
                                  }),
                            ),

                  ],
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class SliderItem extends StatelessWidget {
  final String text;
  final double width;

  const SliderItem({this.text, this.width});

  @override
  Widget build(BuildContext context) {
    return Center(
      child: Container(
        width: width,
        child: Text(
          text,
          textAlign: TextAlign.left,
          style: TextStyle(
          ),
        ),
      ),
    );
  }
}
`

**Comment by Leviosar** [comment]
Any updates on this? I've already implemented a dynamic height on a `StatefulWidget` using the `height` property of `CarouselOptions`, but this is only useful when you know the exactly height of all items (or when you can calculate those heights).

It would be awesome to have the height adapt itself to match item content for variable amounts of text for example.

**Comment by shatanikmahanty** [comment]
I have given solution to this on a stackoverflow question.

[Link to answer](https://stackoverflow.com/a/67700679/12930464).

Let me know if this solves your issue @Leviosar

**Comment by purtellap** [comment]
Something that seemed to work for me: Wrap CarouselSlider in Expanded and set aspectRatio to 1/MediaQuery.of(context).devicePixelRatio)

**Comment by Mijawel** [comment]
I have been having this same issue with a text only carousel and I managed to solve it by using a text painter to work out the height required max height of the carousel (with help from this stack overflow thread https://stackoverflow.com/questions/52659759/how-can-i-get-the-size-of-the-text-widget-in-flutter)

```
double getMaxTextWidgetSize(List<String> texts, double textWidth) {
    double height = 10;
    for (var text in texts) {
      final TextPainter textPainter = TextPainter(
          text: TextSpan(
              text: text,
              style: TextStyle(
                  fontSize: 16.0,
                  fontWeight: FontWeight.w500,
                  color: Colors.black87)),
          maxLines: 1,
          textScaleFactor: MediaQuery.of(context).textScaleFactor,
          textDirection: TextDirection.ltr)
        ..layout(minWidth: 0, maxWidth: double.infinity);
      if (textPainter.size.height > height) {
        final countLines = (textPainter.size.width / textWidth).ceil();
        height = countLines * textPainter.size.height;
      }

    }
    return height;
  }
```

For textWidth, I just approximated the size ratio of my carousel against the total width of the screen and passed that in, which seemed to work fine.

```
double carouselWidthAsFractionOfScreen = 0.66;
getMaxTextWidgetSize(alternatives, MediaQuery.of(context).size.width* carouselWidthAsFractionOfScreen),
```

**Comment by ArkasDev** [comment]
This function would be very helpful.

**Comment by hsul4n** [comment]
I think this is one of most requested feature

**Comment by erperejildo** [comment]
> So any solution?

almost 3 years now. I don't think we are going to have any

**Comment by gustavomaedo** [comment]
My solution is to use GlobalKey() for each index and get the height with key of current index.
`_keys[index].currentContext?.findRenderObject() as RenderBox?)?.size.height ?? 0`

**Comment by jay-b-thakkar** [comment]
> i need custom width for every one of my items.but viewport fraction does not let me to have it.is there a way? [https://github.com/serenader2014/flutter_carousel_slider/issues/397](url)

Same problem with me , not sure if there is any solution / workaround available for this ?

**Comment by ahmet-fennel** [comment]
I tried to update the package to use ExpandablePageView and removed some fields and now it works without specifying a height https://github.com/ahmet-fennel/flutter_carousel_slider

**Comment by krutarth-vaishnav-ups** [comment]
@ahmet-fennel I tried your package and it's working fine without height but "Infinite Scroll" Feature is not working anymore. Can you please look into that?

**Comment by krutarth-vaishnav-ups** [comment]
> I tried to update the package to use ExpandablePageView and removed some fields and now it works without specifying a height https://github.com/ahmet-fennel/flutter_carousel_slider

@ahmet-fennel I tried your package and it's working fine without height but "Infinite Scroll" Feature is not working anymore. Can you please look into that?

**Comment by erperejildo** [comment]
After 4 years, I came back to this and we still have nothing

**Comment by LinXunFeng** [comment]
My solution: [expandable_carousel_slider_demo.dart](https://github.com/fluttercandies/flutter_scrollview_observer/blob/main/example/lib/features/scene/expandable_carousel_slider_demo/expandable_carousel_slider_demo.dart)


https://github.com/user-attachments/assets/cdd99a10-13a7-4d6b-9770-f2e883393ef0

**Comment by kishan-dhankecha** [comment]
Hello @LinXunFeng 

The proposed solution is merely a temporary workaround and lacks practicality due to the requirement for manual height specification. In scenarios where responsive height is necessary, it implies that the sizes of each slide cannot be determined beforehand.

**Comment by erperejildo** [comment]
check second new widget:
https://www.youtube.com/shorts/TyiBYeSI6Po
